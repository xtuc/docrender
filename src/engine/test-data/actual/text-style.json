{
  "title": "publish-bot test - text style",
  "body": {
    "content": [
      {
        "endIndex": 1,
        "sectionBreak": {
          "sectionStyle": {
            "columnSeparatorStyle": "NONE",
            "contentDirection": "LEFT_TO_RIGHT",
            "sectionType": "CONTINUOUS"
          }
        }
      },
      {
        "startIndex": 1,
        "endIndex": 5,
        "paragraph": {
          "elements": [
            {
              "startIndex": 1,
              "endIndex": 5,
              "textRun": {
                "content": "Red\n",
                "textStyle": {
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 1
                      }
                    }
                  }
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 5,
        "endIndex": 11,
        "paragraph": {
          "elements": [
            {
              "startIndex": 5,
              "endIndex": 11,
              "textRun": {
                "content": "Green\n",
                "textStyle": {
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "green": 1
                      }
                    }
                  }
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 11,
        "endIndex": 16,
        "paragraph": {
          "elements": [
            {
              "startIndex": 11,
              "endIndex": 16,
              "textRun": {
                "content": "Blue\n",
                "textStyle": {
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 0.2901961,
                        "green": 0.5254902,
                        "blue": 0.9098039
                      }
                    }
                  }
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 16,
        "endIndex": 17,
        "paragraph": {
          "elements": [
            {
              "startIndex": 16,
              "endIndex": 17,
              "textRun": {
                "content": "\n",
                "textStyle": {
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 0.2901961,
                        "green": 0.5254902,
                        "blue": 0.9098039
                      }
                    }
                  }
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 17,
        "endIndex": 21,
        "paragraph": {
          "elements": [
            {
              "startIndex": 17,
              "endIndex": 21,
              "textRun": {
                "content": "Red\n",
                "textStyle": {
                  "backgroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 1
                      }
                    }
                  },
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 1,
                        "green": 1,
                        "blue": 1
                      }
                    }
                  }
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 21,
        "endIndex": 27,
        "paragraph": {
          "elements": [
            {
              "startIndex": 21,
              "endIndex": 27,
              "textRun": {
                "content": "Green\n",
                "textStyle": {
                  "backgroundColor": {
                    "color": {
                      "rgbColor": {
                        "green": 1
                      }
                    }
                  },
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 1,
                        "green": 1,
                        "blue": 1
                      }
                    }
                  }
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 27,
        "endIndex": 32,
        "paragraph": {
          "elements": [
            {
              "startIndex": 27,
              "endIndex": 32,
              "textRun": {
                "content": "Blue\n",
                "textStyle": {
                  "backgroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 0.2901961,
                        "green": 0.5254902,
                        "blue": 0.9098039
                      }
                    }
                  },
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 1,
                        "green": 1,
                        "blue": 1
                      }
                    }
                  }
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 32,
        "endIndex": 33,
        "paragraph": {
          "elements": [
            {
              "startIndex": 32,
              "endIndex": 33,
              "textRun": {
                "content": "\n",
                "textStyle": {
                  "backgroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 0.2901961,
                        "green": 0.5254902,
                        "blue": 0.9098039
                      }
                    }
                  },
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 1,
                        "green": 1,
                        "blue": 1
                      }
                    }
                  }
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 33,
        "endIndex": 44,
        "paragraph": {
          "elements": [
            {
              "startIndex": 33,
              "endIndex": 44,
              "textRun": {
                "content": "Underlined\n",
                "textStyle": {
                  "underline": true
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 44,
        "endIndex": 49,
        "paragraph": {
          "elements": [
            {
              "startIndex": 44,
              "endIndex": 49,
              "textRun": {
                "content": "Bold\n",
                "textStyle": {
                  "bold": true
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 49,
        "endIndex": 65,
        "paragraph": {
          "elements": [
            {
              "startIndex": 49,
              "endIndex": 65,
              "textRun": {
                "content": "Bold+Underlined\n",
                "textStyle": {
                  "bold": true,
                  "underline": true
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 65,
        "endIndex": 72,
        "paragraph": {
          "elements": [
            {
              "startIndex": 65,
              "endIndex": 72,
              "textRun": {
                "content": "Italic\n",
                "textStyle": {
                  "italic": true
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 72,
        "endIndex": 82,
        "paragraph": {
          "elements": [
            {
              "startIndex": 72,
              "endIndex": 76,
              "textRun": {
                "content": "Big ",
                "textStyle": {
                  "fontSize": {
                    "magnitude": 30,
                    "unit": "PT"
                  }
                }
              }
            },
            {
              "startIndex": 76,
              "endIndex": 82,
              "textRun": {
                "content": "small\n",
                "textStyle": {
                  "fontSize": {
                    "magnitude": 8,
                    "unit": "PT"
                  }
                }
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      }
    ]
  },
  "documentStyle": {
    "background": {
      "color": {}
    },
    "pageNumberStart": 1,
    "marginTop": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginBottom": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginRight": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginLeft": {
      "magnitude": 72,
      "unit": "PT"
    },
    "pageSize": {
      "height": {
        "magnitude": 841.8897637795277,
        "unit": "PT"
      },
      "width": {
        "magnitude": 595.2755905511812,
        "unit": "PT"
      }
    },
    "marginHeader": {
      "magnitude": 36,
      "unit": "PT"
    },
    "marginFooter": {
      "magnitude": 36,
      "unit": "PT"
    },
    "useCustomHeaderFooterMargins": true
  },
  "namedStyles": {
    "styles": [
      {
        "namedStyleType": "NORMAL_TEXT",
        "textStyle": {
          "bold": false,
          "italic": false,
          "underline": false,
          "strikethrough": false,
          "smallCaps": false,
          "backgroundColor": {},
          "foregroundColor": {
            "color": {
              "rgbColor": {}
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          },
          "weightedFontFamily": {
            "fontFamily": "Arial",
            "weight": 400
          },
          "baselineOffset": "NONE"
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "alignment": "START",
          "lineSpacing": 115,
          "direction": "LEFT_TO_RIGHT",
          "spacingMode": "COLLAPSE_LISTS",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "unit": "PT"
          },
          "borderBetween": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderTop": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderBottom": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderLeft": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderRight": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "indentFirstLine": {
            "unit": "PT"
          },
          "indentStart": {
            "unit": "PT"
          },
          "indentEnd": {
            "unit": "PT"
          },
          "keepLinesTogether": false,
          "keepWithNext": false,
          "avoidWidowAndOrphan": true,
          "shading": {
            "backgroundColor": {}
          },
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_1",
        "textStyle": {
          "fontSize": {
            "magnitude": 20,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 20,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 6,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_2",
        "textStyle": {
          "bold": false,
          "fontSize": {
            "magnitude": 16,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 18,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 6,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_3",
        "textStyle": {
          "bold": false,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.2627451,
                "green": 0.2627451,
                "blue": 0.2627451
              }
            }
          },
          "fontSize": {
            "magnitude": 14,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 16,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_4",
        "textStyle": {
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 12,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 14,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_5",
        "textStyle": {
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 12,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_6",
        "textStyle": {
          "italic": true,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 12,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "TITLE",
        "textStyle": {
          "fontSize": {
            "magnitude": 26,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 3,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "SUBTITLE",
        "textStyle": {
          "italic": false,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 15,
            "unit": "PT"
          },
          "weightedFontFamily": {
            "fontFamily": "Arial",
            "weight": 400
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 16,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      }
    ]
  },
  "revisionId": "ALBJ4LuBxIeQdWqVNAtsVIIUKSPYIVZfkpFliDwtY-VNFMXucYrIOzP3uHIKkqsFAJxGWTlthBhbgmiZOIHgB0w",
  "suggestionsViewMode": "SUGGESTIONS_INLINE",
  "documentId": "19IommdjdbNq-CjGg1rtY7fWBf4nQXT1Qce8xSfrv0JA"
}
