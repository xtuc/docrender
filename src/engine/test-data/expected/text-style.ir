(
    content: [
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Red", (
                    styles: [
                        ("color", "rgb(100%, 0%, 0%)"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Green", (
                    styles: [
                        ("color", "rgb(0%, 100%, 0%)"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Blue", (
                    styles: [
                        ("color", "rgb(29.01961%, 52.549023%, 90.98039%)"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Red", (
                    styles: [
                        ("color", "rgb(100%, 100%, 100%)"),
                        ("background-color", "rgb(100%, 0%, 0%)"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Green", (
                    styles: [
                        ("color", "rgb(100%, 100%, 100%)"),
                        ("background-color", "rgb(0%, 100%, 0%)"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Blue", (
                    styles: [
                        ("color", "rgb(100%, 100%, 100%)"),
                        ("background-color", "rgb(29.01961%, 52.549023%, 90.98039%)"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Underlined", (
                    styles: [
                        ("text-decoration", "underline"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Bold", (
                    styles: [
                        ("font-weight", "bold"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Bold+Underlined", (
                    styles: [
                        ("text-decoration", "underline"),
                        ("font-weight", "bold"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Italic", (
                    styles: [
                        ("font-style", "italic"),
                    ],
                )),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Big ", (
                    styles: [
                        ("font-size", "30PT"),
                    ],
                )),
            ], false),
            InlineText([
                TextWithOptions("small", (
                    styles: [
                        ("font-size", "8PT"),
                    ],
                )),
            ], true),
        ]),
        Clear,
    ],
    styles: {
        "pbNORMAL_TEXT": [
            ("line-height", "1.7249999"),
        ],
    },
)