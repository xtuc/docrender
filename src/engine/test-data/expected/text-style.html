<style>
  .pbNORMAL_TEXT {
    line-height: 1.7249999;
  }
</style>
<p><span style="color: rgb(100%, 0%, 0%)">Red</span></p>
<p><span style="color: rgb(0%, 100%, 0%)">Green</span></p>
<p><span style="color: rgb(29.01961%, 52.549023%, 90.98039%)">Blue</span></p>
<p>
  <span
    style="color: rgb(100%, 100%, 100%); background-color: rgb(100%, 0%, 0%)"
    >Red</span
  >
</p>
<p>
  <span
    style="color: rgb(100%, 100%, 100%); background-color: rgb(0%, 100%, 0%)"
    >Green</span
  >
</p>
<p>
  <span
    style="
      color: rgb(100%, 100%, 100%);
      background-color: rgb(29.01961%, 52.549023%, 90.98039%);
    "
    >Blue</span
  >
</p>
<p><span style="text-decoration: underline">Underlined</span></p>
<p><span style="font-weight: bold">Bold</span></p>
<p>
  <span style="text-decoration: underline; font-weight: bold"
    >Bold+Underlined</span
  >
</p>
<p><span style="font-style: italic">Italic</span></p>
<p>
  <span style="font-size: 30pt">Big </span
  ><span style="font-size: 8pt">small</span>
</p>
